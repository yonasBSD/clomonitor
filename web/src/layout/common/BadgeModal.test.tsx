import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Foundation } from 'clo-ui/components/Foundation';
import { vi } from 'vitest';

import BadgeModal from './BadgeModal';

const mockOnCloseModal = vi.fn();

const defaultProps = {
  foundation: Foundation.cncf,
  projectName: 'proj',
  openStatus: { status: true, name: 'badge' },
  onCloseModal: mockOnCloseModal,
};

describe('BadgeModal', () => {
  afterEach(() => {
    vi.resetAllMocks();
  });

  it('creates snapshot', () => {
    const { asFragment } = render(<BadgeModal {...defaultProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  describe('Render', () => {
    it('renders markdown tab', () => {
      render(<BadgeModal {...defaultProps} />);

      expect(screen.getByText('Project badge')).toBeInTheDocument();
      expect(screen.getAllByText('Markdown')).toHaveLength(2);
      expect(screen.getAllByRole('button', { name: /Open tab/ })).toHaveLength(2);

      const badge = screen.getByAltText('CLOMonitor badge');
      expect(badge).toBeInTheDocument();
      const endpointUrl = `https://img.shields.io/endpoint?url=${window.location.origin}/api/projects/cncf/proj/badge`;
      expect(badge).toHaveProperty('src', endpointUrl);
      const code = screen.getByTestId('code');
      expect(code).toBeInTheDocument();
      expect(code).toHaveTextContent(`[![CLOMonitor](${endpointUrl})](${window.location.origin}/projects/cncf/proj)`);
    });

    it('renders ascii tab', async () => {
      render(<BadgeModal {...defaultProps} />);

      expect(screen.getAllByText('AsciiDoc')).toHaveLength(2);
      const btns = screen.getAllByRole('button', { name: /Open tab/ });
      expect(btns[1]).toHaveTextContent('AsciiDoc');
      await userEvent.click(btns[1]);

      const badge = screen.getByAltText('CLOMonitor badge');
      expect(badge).toBeInTheDocument();
      const endpointUrl = `https://img.shields.io/endpoint?url=${window.location.origin}/api/projects/cncf/proj/badge`;
      expect(badge).toHaveProperty('src', endpointUrl);
      const code = screen.getByTestId('code');
      expect(code).toBeInTheDocument();
      expect(code).toHaveTextContent(`${window.location.origin}/projects/cncf/proj[image:${endpointUrl}[CLOMonitor]]`);
    });
  });
});
