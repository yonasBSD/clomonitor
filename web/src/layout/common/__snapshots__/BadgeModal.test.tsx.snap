// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BadgeModal > creates snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="modal-backdrop Modal-module_activeBackdrop__FOECF"
      data-testid="modalBackdrop"
    />
    <div
      aria-modal="true"
      class="modal Modal-module_modal__aD7aP Modal-module_active__tFp3m d-block"
      role="dialog"
    >
      <div
        class="modal-dialog modal-lg modal-dialog-centered modal-dialog-scrollable"
      >
        <div
          class="modal-content rounded-0 border border-3 mx-auto Modal-module_content__dWO-B"
        >
          <div
            class="modal-header rounded-0 d-flex flex-row align-items-center Modal-module_header__ILG9i"
          >
            <div
              class="modal-title h5 m-2 flex-grow-1"
            >
              Project badge
            </div>
            <button
              aria-label="Close"
              class="btn-close"
              type="button"
            />
          </div>
          <div
            class="modal-body p-4 h-100 d-flex flex-column"
          >
            <div
              class="my-3"
            >
              <div>
                <ul
                  class="nav nav-tabs Tabs-module_tabs__QzIkz"
                >
                  <li
                    class="nav-item"
                  >
                    <button
                      aria-label="Open tab markdown"
                      class="btn nav-item rounded-0 lightText Tabs-module_btn__Ifkmk active btn-secondary Tabs-module_active__PTNtG"
                    >
                      <span
                        class="d-none d-sm-block"
                      >
                        Markdown
                      </span>
                      <span
                        class="d-block d-sm-none"
                      >
                        Markdown
                      </span>
                    </button>
                  </li>
                  <li
                    class="nav-item"
                  >
                    <button
                      aria-label="Open tab ascii"
                      class="btn nav-item rounded-0 lightText Tabs-module_btn__Ifkmk"
                    >
                      <span
                        class="d-none d-sm-block"
                      >
                        AsciiDoc
                      </span>
                      <span
                        class="d-block d-sm-none"
                      >
                        AsciiDoc
                      </span>
                    </button>
                  </li>
                </ul>
              </div>
              <div
                class="tab-content mt-4"
              >
                <div
                  class="tab-pane fade show active"
                >
                  <div
                    class="mt-2 mb-4"
                  >
                    <img
                      alt="CLOMonitor badge"
                      src="https://img.shields.io/endpoint?url=http://localhost:3000/api/projects/cncf/proj/badge"
                    />
                  </div>
                  <div
                    class="d-flex flex-row align-items-center pb-2 _codeBlock_c6e1ed"
                    data-testid="code"
                  >
                    <pre
                      class="mb-0 flex-grow-1"
                      style="background-color: var(--bg-code); color: var(--color-font); padding: 1rem; margin-bottom: 0px; flex: 1 1 auto; white-space: pre-wrap; word-break: break-word; overscroll-behavior: contain; width: calc(100% + (-1 * (32px + 1rem)));"
                    >
                      [![CLOMonitor](https://img.shields.io/endpoint?url=http://localhost:3000/api/projects/cncf/proj/badge)](http://localhost:3000/projects/cncf/proj)
                    </pre>
                    <div
                      class="position-relative ms-3"
                    >
                      <button
                        aria-label="Copy badge markdown link to clipboard"
                        class="btn btn-sm btn-primary rounded-0 ButtonCopyToClipboard-module_btn__kHG6G"
                        type="button"
                      >
                        <div
                          aria-hidden="true"
                          class="d-flex flex-row align-items-center"
                        >
                          <svg
                            fill="none"
                            height="1em"
                            stroke="currentColor"
                            stroke-linecap="round"
                            stroke-linejoin="round"
                            stroke-width="2"
                            viewBox="0 0 24 24"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <rect
                              height="13"
                              rx="2"
                              ry="2"
                              width="13"
                              x="9"
                              y="9"
                            />
                            <path
                              d="M5 15H4a2 2 0 0 1-2-2V4a2 2 0 0 1 2-2h9a2 2 0 0 1 2 2v1"
                            />
                          </svg>
                        </div>
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="modal-footer p-3"
          >
            <button
              aria-label="Close modal"
              class="btn btn-sm rounded-0 btn-secondary text-uppercase"
              type="button"
            >
              <div
                class="d-flex flex-row align-items-center"
              >
                <svg
                  class="me-2"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0z"
                    fill="none"
                  />
                  <path
                    d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
                <div>
                  Close
                </div>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
